src/app-layer-detect-proto.c

```diff
@@ -1844,6 +1844,16 @@ bool AppLayerRequestProtocolTLSUpgrade(Flow *f)
     return AppLayerRequestProtocolChange(f, 443, ALPROTO_TLS);
 }
 
+void AppLayerForceProtocolChange(Flow *f, AppProto new_proto)
```

Comment:
can this get a comment explaining the logic of only setting f->alproto to new_proto?

Reply:
alproto_ts and alproto_tc are also set to new_proto

Reply:
adding a comment to the function anyways